src/mgr/DaemonServer.cc

```diff
@@ -267,6 +268,8 @@ bool DaemonServer::ms_dispatch(Message *m)
 
 void DaemonServer::maybe_ready(int32_t osd_id)
 {
+  Mutex::Locker l(lock);
+
   if (!pgmap_ready && reported_osds.find(osd_id) == reported_osds.end()) {
```

Comment:
It seems worth making pgmap_ready an atomic<bool> so that you can avoid taking the lock each time here?  It's only false during mgr startup, right?

Reply:
Done!